### PR TITLE
Add contributing file

### PR DESCRIPTION
**Pull request by nelimee**
This PR adds a CONTRIBUTING file to explain how to contribute to the repository.

It also adds an issue template for simple questions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Contributing code
+
+Code contribution follow a rigid but standard process:
+
+1. Check if there is an issue describing the problem you want to solve or the feature you want to implement in the [issues panel](https://github.com/QCHackers/tqec/issues).
+2. If there is no issue, [create an issue](https://github.com/QCHackers/tqec/issues/new/choose) describing what you want to do.
+3. Assign yourself on the issue.
+4. Write down code and submit a pull-request when you think you fixed the issue. If the changes start to accumulate and the to-be-opened PR is large, open a draft PR to let other people have a look at your code, even if it is not fully finished.
+5. Wait for reviews and iterate with reviewers until the PR is in a satisfactory state.
+6. Merge the PR, well done!
```
Comment (Gistbatch) [inline CONTRIBUTING.md:20]:
```suggestion
4. Fork the repository and create a new branch.
5. Write down the code and submit a pull request when you think you fixed the issue. If the changes start accumulating and the to-be-opened PR is large, open a draft PR to let other people look at your code, even if it still needs to be fully finished.
6. Wait for reviews and iterate with reviewers until the PR is satisfactory.
7. Merge the PR and delete the branch; well done!
```

**Review by Gistbatch — APPROVED** [review]
LGTM, I rephrased a bit.

**Comment by nelimee** [comment]
@KabirDubey I will wait a few days to merge that PR, feel free to make comments / review it in the meantime, your feedback on that would be valuable.

**Comment by KabirDubey** [comment]
Sorry I missed this review, my notification settings were off and have been corrected. This looks good to me, thank you @nelimee!

**Pull request merged**